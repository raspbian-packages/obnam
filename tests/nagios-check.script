#!/bin/sh
# Copyright 2011  Lars Wirzenius
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

set -e

opts="--warn-age=1h --critical-age=1d"

# Backup, and pretend to do it during a particular time.
$SRCDIR/tests/backup --pretend-time="1999-01-01 00:00:00"

# Check there's a warning after one
if ! $SRCDIR/tests/obnam $opts nagios-last-backup-age \
        --pretend-time="1999-01-01 01:00:01"
then
    echo 'correctly returned non-zero exit code'
else
    echo 'incorrectly returned zero exit code' 1>&2
    exit 1
fi

# Check there's an error.
if ! $SRCDIR/tests/obnam $opts nagios-last-backup-age \
        --pretend-time="1999-01-02 00:00:01"
then
    echo 'correctly returned non-zero exit code'
else
    echo 'incorrectly returned zero exit code' 1>&2
    exit 1
fi
