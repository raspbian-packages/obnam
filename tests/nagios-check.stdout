WARNING: backup is old.  last backup was 1999-01-01 00:00:00.
correctly returned non-zero exit code
CRITICAL: backup is old.  last backup was 1999-01-01 00:00:00.
correctly returned non-zero exit code
